File: src/main/java/com/hyungu/whoseturn/controllers/RoomController.java
package com.hyungu.whoseturn.controllers;

import com.hyungu.whoseturn.entities.Member;
import com.hyungu.whoseturn.entities.Room;
import com.hyungu.whoseturn.services.RoomService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/rooms")
public class RoomController {

    @Autowired
    private RoomService roomService;

    @GetMapping
    public List<Room> getAllRooms() {
        return roomService.findAllRooms();
    }

    @GetMapping("/{id}")
    public Room getRoomById(@PathVariable Long id) {
        return roomService.getRoomById(id).orElse(null);
    }

    @PostMapping
    public Room createRoom(@RequestBody Room room) {
        return roomService.saveRoom(room);
    }

    public ResponseEntity<Room> updateRoom(@RequestBody Room room) {
        Room findRoom = getRoomById(room.getId());
        if (findRoom == null) {
            return ResponseEntity.notFound().build();
        }
        findRoom.setName(room.getName());
        findRoom.setMembers(room.getMembers());
        return ResponseEntity.ok(roomService.saveRoom(findRoom));
    }

    public ResponseEntity<Room> changeAdmin(@RequestBody Room room, Member newAdmin) {
        Room findRoom = getRoomById(room.getId());
        findRoom.setAdmin(newAdmin);
        return ResponseEntity.ok(roomService.saveRoom(findRoom));
    }

    @DeleteMapping("/{id}")
    public void deleteRoom(@PathVariable Long id) {
        roomService.deleteRoom(id);
    }

}

File: src/main/kotlin/com/hyungu/whoseturn/controllers/RoomController.kt
package com.hyungu.whoseturn.controllers

import com.hyungu.whoseturn.entities.RoomEntity
import com.hyungu.whoseturn.entities.UserEntity
import com.hyungu.whoseturn.repositories.RoomRepository
import com.hyungu.whoseturn.repositories.UserRepository
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RestController
import java.util.Optional

@RestController(value = "rooms")
class RoomController(
    private val roomRepository: RoomRepository,
    private val userRepository: UserRepository
) {

    @GetMapping
    fun findAllRooms(): Iterable<RoomEntity> = roomRepository.findAll()

    @GetMapping("/{id}")
    fun findRoomById(@PathVariable(name = "id") id: String): Optional<RoomEntity> = roomRepository.findById(id)

    @PostMapping("/{user_id}")
    fun postRoom(@PathVariable(name = "user_id") userId: String, @RequestBody room: RoomEntity): RoomEntity {
        if (userRepository.existsById(userId)) {
            val user: UserEntity = userRepository.findById(userId).orElseThrow();
            // TODO: save "Room"
        }
    }

}